test/AdoNetCore.AseClient.Tests/Integration/AseDataReaderTests.cs

```diff
@@ -1044,5 +1081,33 @@ public static IEnumerable<TestCaseData> GetFieldType_ReturnsNonNullableType_Case
             yield return new TestCaseData("select convert(varbinary, null)", typeof(byte[]));
             yield return new TestCaseData("select convert(varchar, null)", typeof(string));
         }
+
+        [TestCaseSource(nameof(ExecuteReader_ReturnsCorrectRecordsAffected_Cases))]
+        public void ExecuteReader_ReturnsCorrectRecordsAffected(string query, int expected)
```

Comment:
What should happen when one execution contains multiple `insert`/`update`/`delete` statements? Should the resulting `RecordsAffected` be the sum of all counts, or just the last?